### PR TITLE
Cleanup guava, findbugs and javax.annotation-api

### PR DESCRIPTION
**Pull request by gjoranv**

### CODE REVIEW

File: configserver/src/test/java/com/yahoo/vespa/config/server/tenant/TenantTest.java [hunk 1]
```diff
@@ -50,11 +49,9 @@ private Tenant createTenant(String name) throws IOException {
...
     @Test
     public void equals() {
-        new EqualsTester()
-                .addEqualityGroup(t1, t2)
-                .addEqualityGroup(t3)
-                .addEqualityGroup(t4)
-                .testEquals();
+        assertEquals(t1, t2);
+        assertNotEquals(t1, t3);
+        assertNotEquals(t1, t4);
```
Comment (jonmv) [inline configserver/src/test/java/com/yahoo/vespa/config/server/tenant/TenantTest.java:54]:
```suggestion
        assertNotEquals(t1, t4);
        assertNotEquals(t3, t4);
```

**Review by jonmv — APPROVED** [review]
I love it!

**Review by hmusum — APPROVED** [review]
💯

File: configserver/src/test/java/com/yahoo/vespa/config/server/tenant/TenantTest.java (hunk 1 above)
Comment (gjoranv) [reply configserver/src/test/java/com/yahoo/vespa/config/server/tenant/TenantTest.java:54]: Adding the assert to the replacement PR

**Comment by gjoranv** [comment]
Replaced by #22710

**Pull request closed without merging**